pkg/mount/deprecated.go

```diff
@@ -54,20 +54,57 @@ var (
 type (
 	// FilterFunc is a type.
 	// Deprecated: use github.com/moby/sys/mountinfo instead.
-	FilterFunc = mountinfo.FilterFunc
-	// Info is a type.
+	FilterFunc = func(*Info) (skip, stop bool)
+
+	// Info is a type
 	// Deprecated: use github.com/moby/sys/mountinfo instead.
-	Info = mountinfo.Info // Info is deprecated
+	Info struct {
+		ID, Parent, Major, Minor                                  int
+		Root, Mountpoint, Opts, Optional, Fstype, Source, VfsOpts string
+	}
 )
 
 // Deprecated: use github.com/moby/sys/mountinfo instead.
 //nolint:golint
 var (
-	Mounted   = mountinfo.Mounted
-	GetMounts = mountinfo.GetMounts
-
+	Mounted           = mountinfo.Mounted
 	PrefixFilter      = mountinfo.PrefixFilter
 	SingleEntryFilter = mountinfo.SingleEntryFilter
 	ParentsFilter     = mountinfo.ParentsFilter
-	FstypeFilter      = mountinfo.FstypeFilter
+	FstypeFilter      = mountinfo.FSTypeFilter
 )
+
+// GetMounts is a function.
+//
+// Deprecated: use github.com/moby/sys/mountinfo.GetMounts() instead.
+//nolint:golint
+func GetMounts(f FilterFunc) ([]*Info, error) {
```

Comment:
Wasn't sure if we wanted to keep this one backward compatible for now; kept it as a separate commit, but can squash it

Reply:
Since mountinfo filters were introduced recently, I don't think they have lots of users so we can just remove those from this wrapper package, and if `GetMounts` is called with a filter, do `panic("use github.com/moby/sys/mountinfo")` or something like that.

Reply:
Or maybe it's time to kill the whole package